torchdata/csrc/CMakeLists.txt

```diff
@@ -0,0 +1,19 @@
+if(BUILD_S3)
+  message(STATUS "Building S3 IO functionality")
+  find_package(AWSSDK REQUIRED COMPONENTS s3 transfer)
```

Comment:
I was asking the question because we may need to specify the path for AWSSDK here.

Reply:
I really hope AWSSDK can provide static library for users to install rather than installing from source code.

Reply:
I'm looking into building AWS SDK CPP static libraries, and doing some experiments now. Therefore, if official `torchdata` is built with S3 IO functionality, users can just install `torchdata` directly.

Reply:
Do you mind adding another environment variable like `S3_PATH` to cmake. And use it as the `PATH` for `find_package`.
And, please also do a rebase. Then, I think this PR is good to go.

Reply:
Where would you like the `S3_PATH` pointing? 
`aws-sdk-cpp` will build and install as shared/dynamically-linked libraries by default, and the libraries are put at default places. If the anyone cares to build and install in a custom manner, the developer should know to add the path to env `PATH`.

Reply:
I just did a test with static libraries. Adding `-DBUILD_SHARED_LIBS=OFF` when cmaking `aws-sdk-cpp` allows `torchdata` to include static libraries when building the binary. The S3 IO datapipes in the new binary work on my instance without aws-sdk-cpp installed and with the nightly-torch.

Reply:
Rebasing now.

Reply:
What is the size footprint in the case of statical linking?

Reply:
@VitalyFedyunin the size will be from 128K to 2.0M with static libraries.

Reply:
Awesome, that means we can prepackage it into the official distro as soon as we are manylinux compatible,